File: mobile/app/src/main/java/edu/cit/basinillo/portkey/shared/ApiResponse.kt
/* ================================================================== */
/*  PORTKEY — Shared API Response Wrapper (Mobile Vertical Slice)     */
/*  Standard envelope matching backend format. Used by all features.  */
/* ================================================================== */

package edu.cit.basinillo.portkey.shared

import com.google.gson.annotations.SerializedName

data class ApiResponse<T>(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: T?,
    @SerializedName("error") val error: ApiError?,
    @SerializedName("timestamp") val timestamp: String?
)

data class ApiError(
    @SerializedName("code") val code: String?,
    @SerializedName("message") val message: String?,
    @SerializedName("details") val details: Any?
)

File: web/features/shared/api-response.ts
// Shared API types for the PortKey web frontend — vertical slice architecture

export interface ApiError {
  code: string;
  message: string;
  details: Record<string, string> | string | null;
}

export interface ApiResponse<T> {
  success: boolean;
  data: T | null;
  error: ApiError | null;
  timestamp: string;
}
